src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -430,6 +441,23 @@ protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbExc
                 new BufferedReader(new InputStreamReader(partitionInputStream)),
                 CSVFormat.newFormat(delimiter).withQuote(quote).withEscape(escape));
         iterator = parser.iterator();
+
+        try {
+          if (!iterator.hasNext()) {
```

Comment:
No, this ```hasNext()``` was intended to cover the case when the worker only has a partial row with a new line (i.e. something like... "example_incomplete_row\n"). 

But I had to add the try/catch because if the partial record had a truncated quote, this ```hasNext()``` would blow up.